driver-core/src/main/com/mongodb/internal/connection/DefaultClusterFactory.java

```diff
@@ -67,6 +78,30 @@ public Cluster createCluster(final ClusterSettings clusterSettings, final Server
                                  final List<MongoCompressor> compressorList, final @Nullable ServerApi serverApi) {
 
         ClusterId clusterId = new ClusterId();
+        ClusterSettings clusterSettings;
+        ServerSettings serverSettings;
+
+        if (noClusterEventListeners(originalClusterSettings, originalServerSettings)) {
```

Comment:
Since the vast number of clients never register a listener, avoiding thread/queue creation for that case.